Makefile

```diff
@@ -1,58 +1,59 @@
 #* Variables
 SHELL ?= /usr/bin/env bash
-PYTHON ?= python
-PYTHONPATH := `pwd`
-POETRY ?= poetry
-
-#* Poetry
-.PHONY: poetry-install
-poetry-install:
-	curl -sSL https://install.python-poetry.org | $(PYTHON) -
-
-.PHONY: poetry-remove
-poetry-remove:
-	curl -sSL https://install.python-poetry.org | $(PYTHON) - --uninstall
-
-.PHONY: poetry-plugins
-poetry-plugins:
-	$(POETRY) self add poetry-plugin-up
-
-.PHONY: poetry-env
-poetry-env:
-	$(POETRY) config virtualenvs.in-project true
+ACTIVEPYTHON = $(shell which python)
+
+#* UV
+.PHONY: setup
+setup:
+	@if ! command -v uv &> /dev/null; then \
+		curl -LsSf https://astral.sh/uv/install.sh | sh; \
+	else \
+		echo "uv is already installed"; \
+	fi
+
+.PHONY: remove
+remove:
+	@read -p "Are you sure you want to remove uv and all its data? [y/N] " confirm && \
+	if [ "$$confirm" = "y" ]; then \
+		uv cache clean; \
+		rm -r "$$(uv python dir)"; \
+		rm -r "$$(uv tool dir)"; \
+		rm ~/.cargo/bin/uv ~/.cargo/bin/uvx; \
+	else \
+		echo "Operation cancelled."; \
```

Comment:
Consider if we need to add a `exit 1` or something to cause the failure to stop. 
If this target is used as dependency to another, the parent target will continue processing. If the intention is to "skip over", then that is fine.